toolkit/offchain/src/reserve/init.rs

```diff
@@ -283,20 +283,38 @@ async fn script_is_initialized<
 	ctx: &TransactionContext,
 	client: &T,
 ) -> Result<bool, anyhow::Error> {
+	Ok(find_script_utxo(script.id, version_oracle, ctx, client).await?.is_some())
+}
+
+// Finds an UTXO at Version Oracle Validator with Datum that contains
+// * script id of the script being initialized
+// * Version Oracle Policy Id
+pub(crate) async fn find_script_utxo<
```

Comment:
In a separate PR I plan to move it to a new module: `version_oracle` together with `VersionOracleData` from `scripts_data`.